src/include/storage/write_ahead_log/log_writer.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "storage/write_ahead_log/log_io.h"
+
+namespace terrier::storage {
+
+// Forward declaration for class LogManager
+class LogManager;
+/**
+ * A LogWriter is responsible for writing serialized log records out to disk.
+ * @param log_manager pointer to the LogManager
+ */
+class LogWriter {
```

Comment:
`LogWriter` is not a good name, because we already have `BufferedLogWriter` in the system. The name should reflect the fact that we are taking serialized log and persisting to a log file on disk. Presumably in the future there will be different types of this (e.g. streaming log to the network.)

Following this line of thought, maybe `LogConsumer` is a good name for the interface. Then we can prefix this with the actual use case.